### PR TITLE
Secret Manager: Save project file without XML declaration

### PR DESCRIPTION
**Pull request by scottaddie**
The Secret Manager's `init` command adds an XML declaration to the project file. To reproduce, use version 3.0.100 of the .NET Core SDK and follow these steps in a command shell:

1. `dotnet new webapi -o TestApi`
1. `cd TestApi`
1. `dotnet user-secrets init`
1. `cat TestApi.csproj`
1. Notice the first line of the resulting project file:

	```xml
	<?xml version="1.0" encoding="utf-8"?>
	<Project Sdk="Microsoft.NET.Sdk.Web">
	  <PropertyGroup>
	    <TargetFramework>netcoreapp3.0</TargetFramework>
		<UserSecretsId>482446be-899f-4ea8-8714-bbe32e56aa51</UserSecretsId>
	  </PropertyGroup>
	  <ItemGroup></ItemGroup>
	</Project>
	```

This PR omits the XML declaration from the project file by taking a different approach to saving the file.

/cc: @CamSoper

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Thanks for the PR, @scottaddie.
@anurse can you please get somebody to look into this? Thanks!

**Comment by anurse** [comment]
@scottaddie could you add a test to validate this change (and prevent us from regressing it)? There are some tests for the init command in https://github.com/aspnet/AspNetCore/blob/de1feec96a5ba738a497a9556dc09ead680ce832/src/Tools/dotnet-user-secrets/test/InitCommandTest.cs already

I think some pseudo-code like this would cover our bases here:

```csharp
[Fact]
public void InitDoesNotAddXMLDeclaration()
{
    var projectDir = _fixture.CreateProject(null);

    // The CreateProject method always creates 'TestProject.csproj' in the projectDir
    var projectFile = Path.Combine(projectDir, "TestProject.csproj"); 

    new InitCommand(null, null).Execute(MakeCommandContext(), projectDir);

    // Read 'projectFile' and validate that the XML declaration isn't present.
}
```

**Comment by anurse** [comment]
Also, I don't think this is likely to be approved for 3.1. Is there a scenario that is *broken* by this? The bar is fairly high for 3.1 changes. Can you retarget to `master`?

**Comment by anurse** [comment]
Added @halter73 so I'm not the only one reviewing code here ;P

**Comment by halter73** [comment]
Not sure about what release this belongs in, but the change looks good. I agree we should add a test though.

**Comment by scottaddie** [comment]
@anurse I'm unaware of a scenario that it breaks. The current behavior is more of a nuisance.

**Comment by anurse** [comment]
> @anurse I'm unaware of a scenario that it breaks. The current behavior is more of a nuisance.

Yeah, it's not going to meet the 3.1 bar then. We can definitely take it in `master` for 5.0 though!

**Comment by scottaddie** [comment]
@halter73 @anurse Thank you for the feedback. This is ready for another review.

File: src/Tools/dotnet-user-secrets/test/InitCommandTest.cs [hunk 1]
```diff
@@ -88,6 +89,18 @@ public void DoesNotGenerateIdForProjectWithSecretId()
...
+        [Fact]
+        public void DoesNotAddXmlDeclarationToProject()
+        {
+            var projectDir = _fixture.CreateProject(null);
+            var projectFile = Path.Combine(projectDir, "TestProject.csproj");
+
+            new InitCommand(null, null).Execute(MakeCommandContext(), projectDir);
+
+            var projectDocument = XDocument.Load(projectFile);
+            Assert.False(projectDocument.Declaration is XDeclaration);
```
Comment (anurse) [inline src/Tools/dotnet-user-secrets/test/InitCommandTest.cs]: What's the actual value here? Is it `null`? Perhaps an `Assert.Null` would be more appropriate?

**Review by analogrelay — APPROVED** [review]
Minor comment about how the test is done. Looks good!

File: src/Tools/dotnet-user-secrets/test/InitCommandTest.cs (hunk 1 above)
Comment (scottaddie) [reply src/Tools/dotnet-user-secrets/test/InitCommandTest.cs]: Thanks for the suggestion! Changed it.

**Review by halter73 — APPROVED** [review]

**Comment by anurse** [comment]
Couldn't find the "Retry" button to only re-run failing legs. 🤞

**Comment by anurse** [comment]
Oof, this got caught up :). I'll respin and try to get this actually landed :)

**Comment by scottaddie** [comment]
@anurse Ping. Is there anything you need from me to get this merged?

**Comment by anurse** [comment]
Probably just flaky tests? Respun the build.

**Comment by scottaddie** [comment]
@anurse After another test run, it looks like there are still flaky tests. Any ideas?

**Comment by anurse** [comment]
Not much to be done right now but keep spinning a bit. Fortunately the msftbot should auto merge this once we get a green build and I can have AzDO only respin failed builds.

**Comment by scottaddie** [comment]
@anurse We finally have a green build! 🎉 Shouldn't msftbot have merged this?

**Comment by anurse** [comment]
It removed `auto-merge` because you added commits and I forgot to reapply the label (oops!). I can manual merge though!

**Pull request merged**